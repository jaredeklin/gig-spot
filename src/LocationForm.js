import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLocation, loadShows } from './actions';
import { fetchShows } from './cleaners/dataCleaner'

export class LocationForm extends Component {
  constructor() {
    super()
    this.state = { 
      zipCode: '',
      radius: ''
    }

    this.dirtyDate = "2018-04-20T21:00:00"
  }

  cleanDate = (dirtyDate) => {
    const currentDate = new Date()
    const splitString = dirtyDate.split('T')
    const date = splitString[0];
    const splitDate = date.split('-')
    const time = splitString[1];
    const splitTime = time.split(':')
    const cleanT = {
      hour: splitTime[0],
      minute: splitTime[1]
    }
    const cleanD = {
      day: splitDate[2],
      month: splitDate[1],
      year: splitDate[0]
    }
    
    console.log(cleanD, cleanT, currentDate)
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
    this.cleanDate(this.dirtyDate)
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    const { loadShows, setLocation } = this.props;
    const shows = await fetchShows(this.state);

    loadShows(shows);
    setLocation(this.state);


    const path = './main';
    this.props.history.push(path);
  };



  render() {
    return (
      <form onSubmit={this.handleSubmit} className='location-form' id={this.props.id}>
      <input 
        type='text'
        name='zipCode'
        value={this.state.zipCode}
        onChange={this.handleChange}
        placeholder='Zip-code'
      />
      <input
        type='text'
        name='radius'
        value={this.state.radius}
        onChange={this.handleChange}
        placeholder='Radius'
      />
      <button>Submit</button>
      </form>
    );
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    setLocation: (location) => (dispatch(setLocation(location))),
    loadShows: (location) => (dispatch(loadShows(location)))
  };
};

export default withRouter(connect(null, mapDispatchToProps)(LocationForm));
