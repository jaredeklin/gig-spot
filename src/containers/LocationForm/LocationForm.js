import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLocation, loadShows } from '../../actions';
import { fetchShows } from '../../cleaners/fetchShows';

export class LocationForm extends Component {
  constructor() {
    super()
    this.state = { 
      zipCode: '',
      radius: ''
    }  
  };

  // cleanDate = () => {
  //   const dirtyISODate = "2018-04-18T01:20:30"
  //   // const currentDateISO = new Date().toISOString()
  //   // const cleanDate = new Date(dirtyISODate).toLocaleDateString([], {
  //   //   month: 'short',
  //   //   day: 'numeric'
  //   // })
  // }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    const { loadShows, setLocation, history } = this.props;
    const shows = await fetchShows(this.state);

    loadShows(shows);
    setLocation(this.state);
    history.push('./main');
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className='location-form' id={this.props.id}>
      <input 
        type='text'
        name='zipCode'
        value={this.state.zipCode}
        onChange={this.handleChange}
        placeholder='Zip-code'
      />
      <input
        type='text'
        name='radius'
        value={this.state.radius}
        onChange={this.handleChange}
        placeholder='Radius'
      />
      <button>Submit</button>
      </form>
    );
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    setLocation: (location) => (dispatch(setLocation(location))),
    loadShows: (location) => (dispatch(loadShows(location)))
  };
};

export default withRouter(connect(null, mapDispatchToProps)(LocationForm));
