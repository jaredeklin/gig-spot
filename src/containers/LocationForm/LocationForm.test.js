import React from 'react';
import { shallow } from 'enzyme';
import { LocationForm, mapDispatchToProps } from './LocationForm';

describe('LocationForm', () => {
  let wrapper;
  let mockEvent;
  let mockFetchShows;
  let mockHistory;
  let mockState;

  beforeEach(() => {
    mockEvent = { preventDefault: jest.fn() };
    mockFetchShows = jest.fn();
    mockHistory = { push: jest.fn() };
    mockState = { location: 'Denver' };

    wrapper = shallow(
      <LocationForm fetchShows={mockFetchShows} history={mockHistory} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if current city is defined', () => {
    wrapper = shallow(<LocationForm currentLocation={'Denver'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should set the state', () => {
    mockEvent = { target: { value: 'Denver', name: 'location' } };
    const expected = { location: 'Denver' };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should map dispatch to props', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.fetchShows();
    expect(mockDispatch).toHaveBeenCalled();
  });

  describe('handleSubmit', () => {
    it('expect fetchShows to have been called with correct params', () => {
      wrapper.setState(mockState);
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockFetchShows).toHaveBeenCalledWith(mockState.location);
    });

    it('expect history.push() to have been called', () => {
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockHistory.push).toHaveBeenCalled();
    });

    it('should reset state to default', () => {
      const expected = { location: '' };

      wrapper.setState(mockState);
      wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });
  });
});
