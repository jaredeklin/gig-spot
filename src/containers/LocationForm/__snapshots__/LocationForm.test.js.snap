// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationForm
    fetchShows={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="location"
          onChange={[Function]}
          placeholder="City"
          type="text"
          value=""
/>,
        <button>
          Submit
</button>,
      ],
      "className": "location-form",
      "id": undefined,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "location",
          "onChange": [Function],
          "placeholder": "City",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="location"
            onChange={[Function]}
            placeholder="City"
            type="text"
            value=""
/>,
          <button>
            Submit
</button>,
        ],
        "className": "location-form",
        "id": undefined,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "location",
            "onChange": [Function],
            "placeholder": "City",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
